knockknock/slack_sender.py

```diff
@@ -50,60 +55,174 @@ def wrapper_sender(*args, **kwargs):
+                        {
+                            "type": "mrkdwn",
+                            "text": "*Machine name:* {}\n"
+                            "*Main call:* {}\n"
+                            "*Starting date:* {}\n".format(
+                                host_name, func_name, start_time.strftime(DATE_FORMAT)
+                            ),
+                        }
+                    ],
+                },
+            ]
+
+        def _successful_message(func_name, host_name, notification, start_time, value):
+            """Uses Slack blocks to report a successful training run with statistics."""
+            end_time = datetime.datetime.now()
+            training_time = _format_train_time(end_time, start_time)
+            blocks = [
+                {"type": "section", "text": {"type": "mrkdwn", "text": notification}},
+                {"type": "divider"},
+                {
+                    "type": "context",
+                    "elements": [
+                        {
+                            "type": "mrkdwn",
+                            "text": "*Machine name:* {}\n"
+                            "*Main call:* {}\n"
+                            "*Starting date:* {}\n"
+                            "*End date:* {}\n"
+                            "*Training Duration:* {}".format(
+                                host_name,
+                                func_name,
+                                start_time.strftime(DATE_FORMAT),
+                                end_time.strftime(DATE_FORMAT),
+                                training_time,
+                            ),
+                        }
+                    ],
+                },
+            ]
+
+            if value is not None:
+                blocks.append({"type": "divider"})
+                try:
+                    str_value = str(value)
+                    dump["blocks"].append(
+                        {
+                            "type": "section",
+                            "text": {
+                                "type": "mrkdwn",
+                                "text": "*Main call returned value:* {}".format(
+                                    str_value
+                                ),
+                            },
+                        }
+                    )
+                except Exception as e:
+                    blocks.append(
+                        "Couldn't str the returned value due to the following error: \n`{}`".format(
+                            e
+                        )
+                    )
+
+            return blocks
+
+        def _format_train_time(end_time, start_time):
+            """Returns a time delta in format HH:MM:SS"""
+            elapsed_time = end_time - start_time
+            hours, remainder = divmod(elapsed_time.seconds, 3600)
+            minutes, seconds = divmod(remainder, 60)
+            training_time = "{:2d}:{:02d}:{:02d}".format(hours, minutes, seconds)
```

Comment:
I think it's important to keep the `number of days` information. Many of us (myself included) have jobs that last several days.